### DISCUSSION TITLE
Mình đã chuyển sang Fcitx5 thay vì IBus Bamboo và IBus Unikey | Dinh Phu Nguyen

### DISCUSSION THREAD

**Discussion by giscus** [dinhphu28/blog#36] [Comments]
# Mình đã chuyển sang Fcitx5 thay vì IBus Bamboo và IBus Unikey | Dinh Phu Nguyen 

Bộ gõ Tiếng Việt thay thế cho IBus Unikey và IBus Bamboo

https://www.dinhphu28.com/blog/2024/fcitx5-unikey-go-tieng-viet/

**Comment by tuandungb** [comment]
Rất vui vì cùng ý tưởng với bác. Cũng đang dùng Opensuse Tumbleweed KDE, và cũng gặp vấn đề với bộ gõ ở môi trường Wayland. Fcitx5-unikey khi đánh vẫn gạch chân như bình thường. Cũng ko được thân thiện lắm với các môi trường IDE khác nhau. Nếu bamboo phát triển 1 bản folk cho Fcitx thì tốt quá, chứ ibus thì đúng ko phù hợp với KDE.

**Reply by zenfas** [reply]
@tuandungb có fcitx5-bamboo rồi nhé

**Comment by Serena1432** [comment]
Huhu duma em phải login vào GitHub chỉ để muốn nói là cảm ơn chủ tus rất nhiều ạ 😭

Em dùng Arch Linux chứ không phải OpenSUSE nhưng cũng đang dùng KDE và gặp lỗi tương tự, tìm Google sửa cả đống bên iBus thì mãi không được cho đến khi thấy post này và cài thử thì dùng ok luôn, mấy cái lỗi trước đó em hay gặp khi dùng iBus cũng đã không còn nữa.

Nếu có ai dùng Arch Linux mà gặp lỗi tương tự thì fcitx5 cũng có trên AUR nhé, chỉ cần dùng `yay` cài là xong.

**Reply by Serena1432** [reply]
Không sửa được comment nên phải reply nhưng cả fcitx5 lẫn bản fork UniKey đều có trên repo `extra` của Arch Linux chứ không phải AUR, nên chỉ cần dùng `pacman` là có thể cài được

**Comment by brianhuster** [comment]
Mình đang dùng ibus_bamboo trên Kubuntu đây, và chỉ muốn nói là để chuyển chế độ gõ của ibus_bamboo thì bạn nhấn `~` để chuyển chọn chế độ gõ nhé. Kinh nghiệm của mình thì dùng app GUI thì chọn chế độ số 1, TUI thì chọn chế độ số 3

**Reply by Serena1432** [reply]
Không thể nhấn ~ được ý anh, thậm chí nhấn vào biểu tượng của iBus trên taskbar thỉnh thoảng còn không hiện gì như đã nói ở trên post rồi.

**Reply by brianhuster** [reply]
Thế chắc ibus của bạn lỗi hoặc chưa cập nhật gì đấy, hoặc cũng có thể bạn đang dùng Wayland. Nhưng nếu bạn dùng fcitx ổn thì cứ dùng thôi, không cần để ý cái kia nữa

**Comment by goatastronaut0212** [comment]
Bài viết khá hay. Mình hiểu được lý do tại sao chúng ta nên sử dụng fcitx5. Cộng đồng bên fcitx5 có vẻ năng động và sôi nổi hơn đặc biệt là họ cũng tích cưc trò chuyện với các nhà phát triển bên Wayland nữa và online chat bên Telegram.

Quay lại ibus-bamboo thì vấn đề của ibus-bamboo đa phần đến từ các dự án đầu nguồn như ibus. Nhưng bản thân hiện tại vấn đề của ibus-bamboo cũng hơi trầm trong khi hỗ trợ trên Wayland chưa tốt như bên Xorg. Vấn đề chuyển chế độ gõ trên Wayland của ibus-bamboo lý do tại sao nó không hoạt động được là bởi vì thư viện goibus mà ibus-bamboo hiện đang sử dụng có vấn đề vì nó tương tác Dbus thay vì gọi API trực tiếp của Ibus mà các DE/WM khác cho rằng việc gọi như vậy hơi kém bảo mật nên hạn chế nó nhưng bạn cũng có thể bật lại cho phép gọi qua Dbus nhưng nó hơi phức tạp và với cá nhân mình thì nếu có lẽ goibus sẽ cần phải bọc Ibus API trong tương lai thay vì gọi qua Dbus ngoài ra 1 thư viện tương tác với Wayland khác buộc phải tự phát triển thêm mình đoán vậy. Nhưng mà nói đến vấn đề chế độ gõ thực ra là con dao hai lưỡi và chỉ với chế độ 1 hay preedit là bạn có thể đủ xài cho hết tất cả trường hợp, mình nhớ nhà phát triển chính của Fcitx5 có đề cập việc tại sao anh ta không thích ibus-bamboo có các chế độ gõ đó và cá nhân mình với cái mã nguồn đó thì cái việc các app trên đòi hỏi đồ trễ cũng khiến cho việc chế độ gõ đó có thể gây ra 1 vài tình huống mình cũng không đoán trước được. Hãy gõ space khi xài preedit không phải là vấn đề quá to tát với mình.

Dù sao đi nữa thì ibus-bamboo vẫn là bộ gõ ổn với mình trên GNOME. Nếu bảo mình có xài fcitx5 mình cũng không ngần ngại nói mình hiện đang dùng fcitx5-bamboo vì cơ bản trải nghiệm gõ và sự hỗ trợ của nó thực sự tốt hơn ibus-bamboo. Cá nhân mình thấy fcitx5-bamboo vẫn tốt hơn fcitx5-unikey vì đôi khi fcitx5-unikey dừng đột ngột kh6ng thể gõ tiếp chữ tiếng Việt khiến mình phải xóa chữ lại khá mất thời gian.

Thực lòng mình khá là cạn kiệt sức lực mỗi khi nhìn vô mã nguồn ibus-bamboo. Có vẻ như tương lai nếu ai đó muốn phát triển 1 bộ gõ tiếng việt nào đó hãy chọn Fcitx5 và mình có thể giúp đỡ nếu mình cảm thấy hứng thú. Còn bây giờ có vẻ mình sẽ sử dụng và đóng góp cho fcitx5-bamboo 1 thời gian vậy.

**Reply by zenfas** [reply]
> Bài viết khá hay. Mình hiểu được lý do tại sao chúng ta nên sử dụng fcitx5. Cộng đồng bên fcitx5 có vẻ năng động và sôi nổi hơn đặc biệt là họ cũng tích cưc trò chuyện với các nhà phát triển bên Wayland nữa và online chat bên Telegram.
> 
> Quay lại ibus-bamboo thì vấn đề của ibus-bamboo đa phần đến từ các dự án đầu nguồn như ibus. Nhưng bản thân hiện tại vấn đề của ibus-bamboo cũng hơi trầm trong khi hỗ trợ trên Wayland chưa tốt như bên Xorg. Vấn đề chuyển chế độ gõ trên Wayland của ibus-bamboo lý do tại sao nó không hoạt động được là bởi vì thư viện goibus mà ibus-bamboo hiện đang sử dụng có vấn đề vì nó tương tác Dbus thay vì gọi API trực tiếp của Ibus mà các DE/WM khác cho rằng việc gọi như vậy hơi kém bảo mật nên hạn chế nó nhưng bạn cũng có thể bật lại cho phép gọi qua Dbus nhưng nó hơi phức tạp và với cá nhân mình thì nếu có lẽ goibus sẽ cần phải bọc Ibus API trong tương lai thay vì gọi qua Dbus ngoài ra 1 thư viện tương tác với Wayland khác buộc phải tự phát triển thêm mình đoán vậy. Nhưng mà nói đến vấn đề chế độ gõ thực ra là con dao hai lưỡi và chỉ với chế độ 1 hay preedit là bạn có thể đủ xài cho hết tất cả trường hợp, mình nhớ nhà phát triển chính của Fcitx5 có đề cập việc tại sao anh ta không thích ibus-bamboo có các chế độ gõ đó và cá nhân mình với cái mã nguồn đó thì cái việc các app trên đòi hỏi đồ trễ cũng khiến cho việc chế độ gõ đó có thể gây ra 1 vài tình huống mình cũng không đoán trước được. Hãy gõ space khi xài preedit không phải là vấn đề quá to tát với mình.
> 
> Dù sao đi nữa thì ibus-bamboo vẫn là bộ gõ ổn với mình trên GNOME. Nếu bảo mình có xài fcitx5 mình cũng không ngần ngại nói mình hiện đang dùng fcitx5-bamboo vì cơ bản trải nghiệm gõ và sự hỗ trợ của nó thực sự tốt hơn ibus-bamboo. Cá nhân mình thấy fcitx5-bamboo vẫn tốt hơn fcitx5-unikey vì đôi khi fcitx5-unikey dừng đột ngột kh6ng thể gõ tiếp chữ tiếng Việt khiến mình phải xóa chữ lại khá mất thời gian.
> 
> Thực lòng mình khá là cạn kiệt sức lực mỗi khi nhìn vô mã nguồn ibus-bamboo. Có vẻ như tương lai nếu ai đó muốn phát triển 1 bộ gõ tiếng việt nào đó hãy chọn Fcitx5 và mình có thể giúp đỡ nếu mình cảm thấy hứng thú. Còn bây giờ có vẻ mình sẽ sử dụng và đóng góp cho fcitx5-bamboo 1 thời gian vậy.

Cái fcitx5-unikey bị lỗi như bạn nói có debug được không, báo cho team fcitx5 họ fix giúp thôi

**Reply by goatastronaut0212** [reply]
@zenfas có lẽ tính năng suggestion keyword sẽ hứng thú mình hơn và nó có thể ngăn chặn việc đó. Mình cũng không rõ bug đó còn không, mình đã từng thử trên app nào và nó bị nhưng nó đã có 1 thời gian dài rất lâu rồi nếu rảnh thì mình sẽ thử lại hết xem.

**Reply by louiszn** [reply]
Tớ cũng vừa chuyển qua lại GNOME tại tớ gặp khá nhiều vấn đề về hiệu năng về Nvidia trên Plasma Wayland, Xorg cũng chỉ khắc phục được một số tình trạng như drop Fps chứ cũng không giúp được nhiều. Hiện tại GNOME đang hoạt động rất mượt và chưa có một vấn đề nghiêm trọng nào xảy ra. Nhưng mà vấn đề lớn nhất đối với tớ vẫn là việc chọn bộ gõ tiếng Việt... Trên GNOME thì tớ thấy ổn nhất có mỗi ibus-bamboo thôi, nhưng mà cũng chỉ giới hạn ở pre-edit mode chứ không thể sử dụng các mode khác. Thật sự thì tớ cũng quen dùng fcitx5-bamboo rồi, nhưng mà trải nghiệm bên ibus có vẻ không được mượt như bên fcitx5 cho lắm 🥲.

**Reply by goatastronaut0212** [reply]
@louiszn Mình hiểu những gì bạn đang trải qua. Cũng có khá là nhiều khía cạnh tại sao lại như thế và ibus-bamboo tệ. Phần nhiều thì thiếu nhiều người đóng góp, hỗn tạp giữa C, Go và Dbus để tương tác ibus và cũng như thiếu hỗ trợ mạnh từ các dự án phía trên. Nhìn vào fcitx5 số báo cáo lỗi thực sự ít và họ hoạt động rất năng động, họ cũng xài C++ gần như mọi thứ. Mình cũng không hiểu tại sao mọi thứ lại tệ và trầm trọng như vậy so với ibus và các bộ gõ được xây từ Ibus nhưng mọi người vẫn để lại sao cho dự án ibus-bamboo khi fcitx5 thực sự tốt hơn. Suy cho cùng thì phần mềm nguồn mở không quá quan trọng miễn xài được là được rồi. Mình cũng không quá trách nếu ai đó sử dụng fcitx5-unikey hay fcitx5-bamboo vì nó hoạt động quá tốt và ngoài sức tưởng tượng. Có lẽ mình sẽ không đóng góp ibus-bamboo trong 1 thời gian rất dài nữa và sẽ dành nhiều thời gian hơn để đóng góp cho fcitx5-unikey hơn. Thêm 1 vài tính năng với C++ dựa trên những API C++ có sẵn và hoạt động tốt và sẽ nhẹ nhàng hơn là phát minh bánh xe ở ibus-bamboo, trộn C, Go và phải xem những gì hỗn loạn đang xảy ra ở bên Ibus.

P/s. Hôm nay có vẻ mình quá burnout khi xem source code và vấn đề cần giải quyết của ibus-bamboo. Nó tệ hơn những gì mình nghĩ. Có lẽ chuyển sang thành 1 người đóng góp fcitx5 cũng không phải quá tệ lắm.

**Reply by brianhuster** [reply]
Nói vậy là fcitx5-bamboo không hỗ trợ mấy chế độ Forward key phải ko bạn?

**Reply by goatastronaut0212** [reply]
@brianhuster yea. Ngắn gọn là vậy. pre-edit thực ra là đủ dùng rồi.

**Comment by brianhuster** [comment]
Hôm nay mình cũng thử chuyển sang fctx5, nhưng mà cho mình hỏi `fctx5-bamboo` có hỗ trợ chế độ Forward Key (mode 3) như của `ibus-bamboo` không bạn? Nếu có thì làm thế nào để thiết lập? Cảm ơn bạn nhiều!

**Reply by louiszn** [reply]
Theo mình biết thì fcitx5-bamboo chỉ có một chế độ gõ duy nhất thôi. Mình thấy khi mà dùng quen rồi cảm giác gõ với preedit vẫn thích hơn nhiều với ổn định hơn trên Wayland á nên bạn cũng có thể thử.

**Comment by nghiata** [comment]
Have a nice day bros!
I'm glad to find you there for a solution for Vietnamese lang in Arch linux (KDE).

I'm sure that I installed all useful packages. However I still can not typing Vietnamese both fcitx5-unikey and fcitx5-bamboo on Plasma 6 KDE.
Have any idea about that?

**Reply by dinhphu28** [reply]
You see it in the System Settings -> Virtual Keyboard, Input Methods or like so, don't u?
